tools/lsp/server_loop.rs

```diff
@@ -1028,20 +1028,30 @@ fn get_document_symbols(
             .filter_map(|child| {
                 let e = child.borrow();
                 Some(DocumentSymbol {
-                    range: make_range.map_range(e.node.as_ref()?.text_range()),
-                    selection_range: make_range
+                    range: offset_mapper.map_range(e.node.as_ref()?.text_range()),
+                    selection_range: offset_mapper
                         .map_range(e.node.as_ref()?.QualifiedName().as_ref()?.text_range()),
                     name: e.base_type.to_string(),
                     detail: (!e.id.is_empty()).then(|| e.id.clone()),
                     kind: lsp_types::SymbolKind::VARIABLE,
-                    children: gen_children(child, ds, make_range),
+                    children: gen_children(child, ds, offset_mapper),
                     ..ds.clone()
                 })
             })
             .collect::<Vec<_>>();
         (!r.is_empty()).then(|| r)
     }
 
+    r.sort_by(|a, b| {
```

Comment:
Why is it not sorted in the first place? We should be traversing the children in order.

Reply:
Ah, i see, indeed.

> I would really like to add more information like struct members and such :-) It's a bit tricky to find all the proper ranges to fill in though.

All the information should be in the done.  You can assume the node is a `syntax_nodes::ObjectType` and then you have the `ObjectTypeMember`  within it.

Why do you want this information though?
Now, i don't think we should have properties and members in the outline itself.

Reply:
> Why do you want this information though? Now, i don't think we should have properties and members in the outline itself.

I agree with Olivier on this. The more information there is in the outline, the deeper it is, the harder it is to read and navigate. That is what I think the primary purpose of the outline is: A compass that gives you a view of the hierarchy of the UI and (by interactivity) a way to navigate it.

Reply:
Right, the outline should contain the whole document :-)

> Yes, I noticed. The ranges are there, but you need to do some parsing to get the name/Type out.

You probably have a different definition of parsing than I do.

```rust
node.ObjectTypeMember().map(|f| {
   let name = parser::identifier_text(&f)?;
   let type = fields.get(name)?;
   let range = f.text_range();
   //...
})
```